testctrl/proto/test_sessions.proto

```diff
@@ -0,0 +1,178 @@
+  // identifier to the session.
+  //
+  // The unique identifier can be used to poll for the session's status and
+  // results.  Streaming is not supported at present.
+  //
+  // It can also be used to cancel while in progress, but cancellations operate
+  // like killing a running process. If they occur while results are being
+  // reported, the persistent storage may receive some but not all of them.
+  //
+  // For the specification of a google.longrunning.Operation message, see the
+  // Long-running AIP at https://aip.dev/151.
+  rpc StartTestSession(StartTestSessionRequest) returns (google.longrunning.Operation) {
+    option (google.longrunning.operation_info) = {
+      response_type: "TestSessionResults"
+      metadata_type: "TestSessionMetadata"
+    };
+  }
+}
+
+message Component {
+  enum Kind {
+    // No kind was specified, so the service will attempt to infer one.
+    //
+    // This inference works as follows:
+    //   1. If this field belongs to the driver_container of the
+    //      StartTestSessionRequest, this kind is inferred to be a driver.
+    //   2. Otherwise, this field is inferred to be a client worker.
+    UNSPECIFIED = 0;
+
+    // Orchestrates the test amongst workers and reports results.
+    DRIVER = 1;
+
+    // Processes incoming requests from client containers.
+    SERVER = 2;
+
+    // Sends outgoing requests to a server container.
+    CLIENT = 3;
+  }
+
+  // The container image and a version as it appears in a docker registry.
+  //
+  // When using GCR, this will likely be a fully-qualified URL and either an
+  // explicit tag or sha256 hash. For example, a Java worker image may be
+  // similar to one of the following:
+  //
+  //   - gcr.io/grpc-testing/java_worker:v3.2.19
+  //   - gcr.io/grpc-testing/java_worker@sha256:82b6360a84c19f23ed7ee9...
+  //
+  // The "latest" tag is automatically set by the registry, and there
+  // are no guarantees that it will point to a specific image. It should be
+  // avoided.
+  string container = 1;  // [(google.api.field_behavior) = REQUIRED];
+
+  // The kind which assigns the responsibilities of this container.
+  Kind kind = 2;  // [(google.api.field_behavior) = OPTIONAL];
+
+  // The number of identical components that should be created with these
+  // settings. For example, a "client" component can be configured once and
+  // specify "2" replicas. Two identical versions will be available for tests.
+  //
+  // This field is ignored for driver components. It may eventually be supported
+  // on server components, but this is not yet available.
+  int32 replicas = 3;
+}
+
+message StartTestSessionRequest {
+  // The test scenarios to run as a part of the test session.
+  repeated grpc.testing.Scenario scenarios = 1;  // [(google.api.field_behavior) = REQUIRED];
+
+  // The component which orchestrates running the test scenarios amongst workers.
+  Component driver = 2;  // [(google.api.field_behavior) = OPTIONAL];
```

Comment:
This field could be joined with workers (below), but there needs to be something to orchestrate the test. Therefore, it's required. I feel like it makes sense to split this into its own field. Thoughts?